src/Npgsql/Replication/INpgsqlReplicationMessage.cs

```diff
@@ -0,0 +1,31 @@
+﻿using JetBrains.Annotations;
+using NpgsqlTypes;
+using System;
+
+namespace Npgsql.Replication
+{
+    /// <summary>
+    /// The common base interface for all streaming replication messages
+    /// </summary>
+    [PublicAPI]
+    public interface INpgsqlReplicationMessage
```

Comment:
> NOTE: I see that XLogDataMessage is a struct, so can implement an interface but not extend a base class.

That was the Idea, but if we go into the recycling business (which I think is a good idea because our implementation of the Logical Streaming Replication Protocol does loads of short-lived allocations) you are right.

Reply:
Done in 461ea0596af360268025d4dcb4618b73eb79c7d5